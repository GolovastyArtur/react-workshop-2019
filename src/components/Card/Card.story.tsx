import React from 'react';

import {storiesOf} from '@storybook/react';

import {Card} from './Card';
import {card, cardWithoutImage} from './fixtures/card';

storiesOf('Card', module)
    .add('default', () => (
        <Card card={card} />
    ))
    .add('no image', () => (
        <Card card={cardWithoutImage} />
    ));
