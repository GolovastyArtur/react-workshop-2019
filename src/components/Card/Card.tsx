import * as React from 'react';
import {CardSchema} from '../../schema/CardSchema';
import {getCardImage} from '../../helpers/getCardImage/getCardImage';
import {Image} from '../Image/Image';

import './Card.css';
import {getCardImageRatio} from '../../helpers/getCardImageRatio/getCardImageRatio';

export interface CardProps {
    card: CardSchema;
}

export const Card: React.FC<CardProps> =
    function Card({card}: CardProps) {
        let imageUrl = getCardImage(card);
        let ratio = getCardImageRatio(card);

        return (
            <div className="card">
                <Image
                    src={imageUrl}
                    alt={card.description}
                    ratio={ratio}
                    className="card__image"
                />

                <div className="card__profile">
                    <div className="card__close">
                        <button className="close">
                            <span className="close__icon" />
                        </button>
                    </div>

                    <div className="person person_card">
                        <a
                            href="https://yandex.ru/collections/user/drexxar/"
                            target="_blank"
                            aria-label="text"
                            className="person__avatar"
                            title="Сергей Глушков"
                            rel="noopener noreferrer"
                        >
                            <img
                                className="avatar"
                                src="https://avatars.mds.yandex.net/get-yapic/27232/591039997-1555108484/islands-retina-small"
                                alt="Сергей Глушков"
                            />
                        </a>

                        <div className="person__info">
                            <div className="board">
                                <a
                                    href="https://yandex.ru/collections/user/drexxar/modnaia-boroda/"
                                    className="board__link"
                                    target="_blank"
                                    rel="noopener noreferrer"
                                >
                                    Модная борода
                                </a>
                            </div>

                            <div className="user">
                                <a
                                    className="user__link"
                                    href="/collections/user/drexxar/"
                                    target="_blank"
                                >
                                    Сергей Глушков
                                </a>
                            </div>

                            <div className="date">
                                <span className="date__desc">15 дней назад</span>
                            </div>
                        </div>
                    </div>

                    <div className="card__article article">
                        <div className="article__desc">
                            Как выбрать правильную форму бороды, как отрастить её и стоит ли девушке относиться
                            к ней негативно.
                        </div>

                        <a
                            className="article__link"
                            href="http://bowandtie.ru/vse-chto-nuzhno-znat-o-borode-ili-oda-bradosti/"
                            target="_blank"
                            rel="noopener noreferrer"
                        >
                            bowandtie.ru
                        </a>
                    </div>
                </div>
            </div>
        );
    };
