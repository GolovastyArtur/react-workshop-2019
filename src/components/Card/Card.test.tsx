import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {Card} from './Card';
import {card, cardWithoutImage} from './fixtures/card';

describe('Card', function () {
    it('render default', function() {
        let component = renderer.create(<Card card={card} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('render no image', function() {
        let component = renderer.create(<Card card={cardWithoutImage} />);

        expect(component.toJSON()).toMatchSnapshot();
    });
});
