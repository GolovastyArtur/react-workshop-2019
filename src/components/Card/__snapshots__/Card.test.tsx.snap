// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render default 1`] = `
<div
  className="card"
>
  <div
    className="image card__image image_hidden"
    style={
      Object {
        "minHeight": "150vw",
      }
    }
  >
    <img
      alt="Скандинавский стиль в интерьере кухни в голубых оттенках"
      className="image__image"
      onLoad={[Function]}
      src="https://avatars.mds.yandex.net/get-pdb/27625/7c81ce06-f70e-405b-87b4-fe826ba9362e/s375"
    />
  </div>
  <div
    className="card__profile"
  >
    <div
      className="card__close"
    >
      <button
        className="close"
      >
        <span
          className="close__icon"
        />
      </button>
    </div>
    <div
      className="person person_card"
    >
      <a
        aria-label="text"
        className="person__avatar"
        href="https://yandex.ru/collections/user/drexxar/"
        rel="noopener noreferrer"
        target="_blank"
        title="Сергей Глушков"
      >
        <img
          alt="Сергей Глушков"
          className="avatar"
          src="https://avatars.mds.yandex.net/get-yapic/27232/591039997-1555108484/islands-retina-small"
        />
      </a>
      <div
        className="person__info"
      >
        <div
          className="board"
        >
          <a
            className="board__link"
            href="https://yandex.ru/collections/user/drexxar/modnaia-boroda/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Модная борода
          </a>
        </div>
        <div
          className="user"
        >
          <a
            className="user__link"
            href="/collections/user/drexxar/"
            target="_blank"
          >
            Сергей Глушков
          </a>
        </div>
        <div
          className="date"
        >
          <span
            className="date__desc"
          >
            15 дней назад
          </span>
        </div>
      </div>
    </div>
    <div
      className="card__article article"
    >
      <div
        className="article__desc"
      >
        Как выбрать правильную форму бороды, как отрастить её и стоит ли девушке относиться к ней негативно.
      </div>
      <a
        className="article__link"
        href="http://bowandtie.ru/vse-chto-nuzhno-znat-o-borode-ili-oda-bradosti/"
        rel="noopener noreferrer"
        target="_blank"
      >
        bowandtie.ru
      </a>
    </div>
  </div>
</div>
`;

exports[`Card render no image 1`] = `
<div
  className="card"
>
  <div
    className="card__profile"
  >
    <div
      className="card__close"
    >
      <button
        className="close"
      >
        <span
          className="close__icon"
        />
      </button>
    </div>
    <div
      className="person person_card"
    >
      <a
        aria-label="text"
        className="person__avatar"
        href="https://yandex.ru/collections/user/drexxar/"
        rel="noopener noreferrer"
        target="_blank"
        title="Сергей Глушков"
      >
        <img
          alt="Сергей Глушков"
          className="avatar"
          src="https://avatars.mds.yandex.net/get-yapic/27232/591039997-1555108484/islands-retina-small"
        />
      </a>
      <div
        className="person__info"
      >
        <div
          className="board"
        >
          <a
            className="board__link"
            href="https://yandex.ru/collections/user/drexxar/modnaia-boroda/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Модная борода
          </a>
        </div>
        <div
          className="user"
        >
          <a
            className="user__link"
            href="/collections/user/drexxar/"
            target="_blank"
          >
            Сергей Глушков
          </a>
        </div>
        <div
          className="date"
        >
          <span
            className="date__desc"
          >
            15 дней назад
          </span>
        </div>
      </div>
    </div>
    <div
      className="card__article article"
    >
      <div
        className="article__desc"
      >
        Как выбрать правильную форму бороды, как отрастить её и стоит ли девушке относиться к ней негативно.
      </div>
      <a
        className="article__link"
        href="http://bowandtie.ru/vse-chto-nuzhno-znat-o-borode-ili-oda-bradosti/"
        rel="noopener noreferrer"
        target="_blank"
      >
        bowandtie.ru
      </a>
    </div>
  </div>
</div>
`;
